packages/common/address.ts

```diff
@@ -25,8 +25,39 @@ export interface AddressInputParts {
   unit?: string
 }
 
+/** Returns a formatted unit, using pound (#) only when necessary */
+export const formatUnit = (unit: string) => {
+  // Users might use pounds erroneously, let this function decide whether
+  // to use it or not.
+  const normalized = unit.replace(/#/g, '')
+
+  // Searchs for any abbreviation that don't require pounds
+  //
+  // https://pe.usps.com/text/pub28/28apc_003.htm#ep538629
+  const dontUsePound = /\b(APT|BLDG|DEPT|FL|FRNT|HNGR|KEY|LBBY|LOT|LOWR|OFC|PH|PIER|REAR|RM|SIDE|SLIP|SPC|STOP|STE|TRLR|UNIT|UPPR)\b/g
+  return !normalized.toUpperCase().match(dontUsePound)
+    ? `# ${normalized}`
+    : normalized
+}
+
 export const formatAddressInputParts = (addr: AddressInputParts): string => {
   return addr.unit
-    ? `${addr.street} #${addr.unit}, ${addr.city}, ${addr.state} ${addr.postcode}`
+    ? `${addr.street} ${formatUnit(addr.unit)}, ${addr.city}, ${addr.state} ${addr.postcode}`
     : `${addr.street}, ${addr.city}, ${addr.state} ${addr.postcode}`
 }
+
+export const splitStreetAndNumber = (street: string): {
+  street: string
+  number: string | null
+} => {
+  // Finds only numbers between spacing or at the end/beginning of sentences,
```

Comment:
Why can't we just test for leading numbers (including ignoring initial spaces)?  I think it's OK if the number is blank (e.g. US Congress has an annoying mailing address of "Ford House Office Building": https://www.house.gov/doing-business-with-the-house/leases/contact-us).

Let's get these cases as well: "123 W Midlands Dr" or "184 E John Street".

We also want to not reject fractional addresses (see https://nypost.com/2018/11/07/the-origins-of-new-york-citys-mysterious-fractional-addresses/).  For example, let's try to get "267 1/2 Water St" and "267.5 Water St" using narrowly tailored regex.  Add this NYPost article for context in the unit tests.

Also, "sentences" should be "address".